modules/tinymce/src/core/main/ts/newline/InsertBlock.ts

```diff
@@ -420,11 +421,15 @@ const insert = (editor: Editor, evt?: EditorEvent<KeyboardEvent>): void => {
   // Enter inside block contained within a LI then split or insert before/after LI
   if (containerBlockName === 'LI' && !ctrlKey) {
     const liBlock = containerBlock as HTMLLIElement;
-    parentBlock = liBlock;
+
     containerBlock = liBlock.parentNode;
     parentBlockName = containerBlockName;
   }
 
+  if (NodeType.isElement(containerBlock) && InsertAccordion.shouldPreventInsertParagraph(editor, shiftKey, parentBlock)) {
```

Comment:
This is not really preventing inserting a paragraph since it later inserts a paragraph. So maybe it should be called `isAtEmptyBlockAtEndOfDetails` or something.

Reply:
Renamed to `isLastEmptyBlockInAccordion`. If we'll decide to rename everything to `details` instead of `accordion`, then will rename it once again.